Realm/RLMQueryUtil.mm

```diff
@@ -42,11 +42,14 @@
 NSString * const RLMPropertiesComparisonTypeMismatchReason = @"Property type mismatch between %@ and %@";
 NSString * const RLMUnsupportedTypesFoundInPropertyComparisonReason = @"Comparison between %@ and %@";
 
-namespace {
+// check a precondition and throw an exception if it is not met
+// this should be used if the condition being false indicates a bug in the caller
+// of the function checking its preconditions
+void RLMPrecondition(bool condition, NSString *name, NSString *format, ...) {
```

Comment:
This change is incorrect. The code is structured in the way that it was so that the conditional would be inlined in the caller, but the error path would not be. RLMKeyPathFromString should just not use RLMPrecondition as making it usable from multiple source files without defeating the goals of it would be a bit complex and using it for the single check in that function isn't particularly useful compared to just a normal check.